src/Monarch/VirtualDom/OutputHandlerTree.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Output handler tree ADT
+ */
+export type OutputHandlerTree = OutputHandlerTree.Root | OutputHandlerTree.Node
```

Comment:
`OutputHandlerTree` data type is not a tree. It's just linked list of nodes that change the output from current node backward to the root.

Reply:
![sooriland_stickerpack_agadyvkaaj8smwu](https://user-images.githubusercontent.com/20781126/101389622-f2a52c80-38d6-11eb-917c-82332f8e3cb2.png)

Reply:
Nice catch. It's kinda a `NonEmpty` but there is a difference between first element and others type. The first element is message dispatcher and should be of type `message -> unit` and others are message transformer and should be of type `messageA -> messageB`. So I think it's reasonable to introduce a new datatype for it.